doc/developer-guide/src/main/markdown/code-guide.md

```diff
@@ -0,0 +1,446 @@
+   of the source file. Inner class definitions should be indented 4 spaces more than their enclosing class [mandatory].
+7. Declare a class as final only if specialisation will never be required and improved performance is essential.
+   With modern JVMs there in fact may be no performance advantage. Warning: use of final limits code reuse [mandatory].
+8. For all but simplest classes the following methods should have useful definitions [recommended]:
+    ```java
+        public boolean equals(Object obj)
+        public int hashCode()
+        public String toString()
+    ```
+9. The order of presentation of the sections in a class should be [mandatory]:
+   * Variables
+   * Methods
+
+#### Variables
+This section gives guidelines for class and instance variable definitions in Java.
+In this section if a rule uses the term variable rather than instance variable or class variable,
+then the rule applies to both types of variable.
+
+1. The order of presentation of variables in a class definition should be [recommended]:
+    * private, protected, public: static final variables (aka constant class variables).
+    * private, protected, public: static variables (aka class variables).
+    * private, protected, public: final variables (aka constant instance variables).
+    * private, protected, public: variables (aka instance variables).
+    It should be noted that as javadoc will automatically order variables in a consistent manner,
+    rigid adherence to this rule is not necessary.
+2. Variable modifiers should be presented in the following order: static, final, transient, volatile [mandatory].
+3. The names of static final variables should be upper case with subsequent words prefixed with an underscore [mandatory].
+   For example:
+    ```java
+    public static final int NOT_FOUND = -1;
+    ```
+4. When a subclass refers to a static final variable defined in a parent class, access should be qualified
+    by specifying the defining class name [mandatory].
+    For example: use `ParentClass.MAX` rather than `MAX`.
+5. The names of variables (other that static final) should start with a lower case letter.
+   Any words that are contained in the rest of the variable name should be capitalised [mandatory].
+    For example:
+    ```java
+    String name;
+    String[] childrensNames;
+     ```
+6. Class and instance variables must be prefixed with an underscore (_) [mandatory].
+7. Variables must not be named using the so-called Hungarian notation [mandatory].
+   For example:
+    ```java
+    int nCount = 4; // not allowed
+    ```
+8. Only one variable may be defined per line [mandatory].
+9. Variable declarations should be indented 4 spaces more than their enclosing class [mandatory].
+10. All variables should be preceded by a javadoc comment that specifies what the variable is for,
+    where it is used and so forth. The comment should be of the following form and be indented
```

Comment:
I actually re-phrased the rule. Adding comments for field description is an old way of code development.  The field name should be self explanatory.  The comments are only needed when the implementation decision is not obvious from the code. Otherwise, they should be avoided.